processing/src/main/java/io/druid/segment/SimpleQueryableIndex.java

```diff
@@ -63,8 +68,20 @@ public SimpleQueryableIndex(
         columnNamesBuilder.add(column);
       }
     }
+    String[] dimArr = new String[dimNames.size()];
+    int index = 0;
+    for (String dim : dimNames) {
+      if (Strings.isNullOrEmpty(dim)) {
+        log.warn("Null or Empty Dimension found");
+        continue;
+      }
+      dimArr[index++] = dim;
+    }
+    dimArr = Arrays.stream(dimArr).filter(d -> !Strings.isNullOrEmpty(d))
```

Comment:
This filter is not necessary, since null/empty values are already being filtered out in the for loop above.

Reply:
Removed